Fixes #18978 - Allow filtering of Interfaces in the GUI by 802.1Q Mode

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes #18978 - Allow filtering of Interfaces in the GUI by 802.1Q Mode

This PR enhances VLAN filtering and filter form usability for the DCIM and Virtualization modules by adding VLAN mode filtering, updating tests, and refactoring field organization.

#### Changes

- **VLAN Mode Filtering:**
  - Added filtering by 802.1Q VLAN mode to Common Interface (DCIM and VMInterface) FilterSets.  
  - Updated tests to accept a list for the mode parameter.

- **Refactoring:**
  - Split the 'Attributes' FieldSet into 'Attributes' and 'Addressing' for improved clarity.

#### Impact

These improvements allow users to efficiently filter interfaces by VLAN mode.
I opted for a `MultipleChoiceFilter` to maintain consistency with how PoE mode filtering is handled, even though this **introduces a breaking change**.

Thank you for reviewing this PR!

pheus committed:
feat(dcim): Add VLAN mode filter to CommonInterface

Introduces a new FilterSet for VLAN mode in CommonInterfaceFilterSet.
This allows filtering interfaces based on their VLAN mode using defined
choices.

pheus committed:
feat(dcim): Add VLAN mode filter to Interface FilterForm

Add a field to InterfaceFilterSet to filter interfaces by 802.1Q VLAN
mode.

pheus committed:
feat(virtualization): Add VLAN mode filter to VMInterface

Add a field to VMInterfaceFilterSet to filter interfaces by 802.1Q VLAN
mode.

pheus committed:
fix(dcim): Correct mode filter parameter type in tests

Updates the `mode` filter parameter to accept a list instead of a single
value in `test_filtersets.py`. Ensures proper count assertion for
accurate test behavior.

pheus committed:
feat(virtualization): Add tests for VLAN mode filtering

Introduces tests to validate filtering by `mode` for VMInterface.
Ensures correct filtering for 802.1Q VLAN mode.

jeremystretch commented on netbox/dcim/forms/filtersets.py:
```diff
@@ -1403,6 +1404,16 @@ class InterfaceFilterForm(PathEndpointFilterForm, DeviceComponentFilterForm):
+        required=False,
+        label=_('802.1Q mode')
+    )
+    vlan_translation_policy_id = DynamicModelMultipleChoiceField(
```
We'll have to omit the filter for VLAN translation policy from this PR, as it wasn't in scope for FR #18978. (However, you're welcome to submit a separate FR for it.)

pheus replied:
Thanks for the feedback. I will remove the unrelated changes and open a separate FR.

pheus commented on netbox/virtualization/forms/filtersets.py:
```diff
@@ -200,7 +201,9 @@ class VMInterfaceFilterForm(NetBoxModelFilterSetForm):
         FieldSet('cluster_id', 'virtual_machine_id', name=_('Virtual Machine')),
-        FieldSet('enabled', 'mac_address', 'vrf_id', 'l2vpn_id', name=_('Attributes')),
+        FieldSet('enabled', name=_('Attributes')),
+        FieldSet('vrf_id', 'l2vpn_id', 'mac_address', name=_('Addressing')),
```
@jeremystretch, would you prefer I keep the original grouping of the Attributes FieldSet, or is the split into separate Attributes and Addressing FieldSets something you’d be interested in keeping?

I’m happy to revert it.

jeremystretch replied:
Either way is fine by me.

pheus committed:
refactor(virtualization): Reorganize FieldSets in FilterSets

Splits the 'Attributes' FieldSet into two distinct FieldSets for better
clarity: 'Attributes' and 'Addressing'. This improves form organization
and makes it more intuitive for users.

pheus commented:
I've removed the commits related to VLAN Translation Policy filtering and kept the split of the FieldSet as discussed.
Thank you very much for your time, review, and patience! I really appreciate it!

jeremystretch reviewed (approved):
Thanks @pheus!
